### PR TITLE
Bypass vbft msg and  Remove peer which have error occur in connecting state.

### PR DESCRIPTION
**Pull request by BooniesFX**
Bypass vbft msg actor under other consensus algo
Remove peer which have error occur in connecting state.

Signed-off-by: Xiang Fu <fuxiang@onchain.com>

### CODE REVIEW

File: p2pserver/message/utils/msg_handler.go [hunk 1]
```diff
@@ -259,12 +262,14 @@ func VersionHandle(data *msgCommon.MsgPayload, p2p p2p.P2P, pid *evtActor.PID, a
...
 			remotePeer.CloseCons()
 			p2p.RemovePeerConsAddress(data.Addr)
+			p2p.RemoveFromConnectingList(data.Addr)
 			return errors.New("The node handshake with itself ")
 		}
 
 		s := remotePeer.GetConsState()
 		if s != msgCommon.INIT && s != msgCommon.HAND {
 			log.Warn("Unknown status to received version", s)
+			p2p.RemoveFromConnectingList(data.Addr)
```
Comment (laizy) [inline p2pserver/message/utils/msg_handler.go]: you may also need close the net connection .

Comment (BooniesFX) [reply p2pserver/message/utils/msg_handler.go]: clean remove logic, close connection contain same action
if dual port not support, set consensus port to zero which avoid be connected by other peer

File: p2pserver/message/msg_pack/msg_pack.go [hunk 2]
```diff
@@ -208,12 +208,15 @@ func NewVersionPayload(n p2pnet.P2P, isCons bool, height uint32) mt.VersionPaylo
...
 		Services:     n.GetServices(),
 		SyncPort:     n.GetSyncPort(),
-		ConsPort:     n.GetConsPort(),
 		Nonce:        n.GetID(),
 		IsConsensus:  isCons,
 		HttpInfoPort: n.GetHttpInfoPort(),
 	}
-
+	if config.DefConfig.P2PNode.DualPortSupport {
+		vpl.ConsPort = n.GetConsPort()
```
Comment (Honglei-Cong) [inline p2pserver/message/msg_pack/msg_pack.go]: why not set ConsPort right on p2p module initialisation, so n.GetConsPort() returns expected answer?

File: p2pserver/message/utils/msg_handler.go [hunk 3]
```diff
@@ -441,16 +436,17 @@ func VerAckHandle(data *msgCommon.MsgPayload, p2p p2p.P2P, pid *evtActor.PID, ar
 				}
 				nodeConsensusAddr := addr[:i] + ":" +
 					strconv.Itoa(int(remotePeer.GetConsPort()))
-				go p2p.Connect(nodeConsensusAddr, true)
+				if remotePeer.GetConsPort() > 0 {
```
Comment (Honglei-Cong) [inline p2pserver/message/utils/msg_handler.go]: change the upper if as:
       if config.DefConfig.P2PNode.DualPortSupport && remotePeer.GetConsPort() > 0

is 'return nil' ok "if i < 0"?
the following p2p.Send(msgpay.NewAddrReq()) is not required?

File: p2pserver/message/msg_pack/msg_pack.go (hunk 2 above)
Comment (BooniesFX) [reply p2pserver/message/msg_pack/msg_pack.go]: good idea, logic added in netserver

File: p2pserver/message/utils/msg_handler.go (hunk 3 above)
Comment (BooniesFX) [reply p2pserver/message/utils/msg_handler.go]: refer to https://github.com/ontio/ontology/pull/178, the handler logic would remove error value.so all return error will delete in that pull.

NewAddrReq is needed for acquiring nbr node`s address.
other changes completed in latest code.

**Review by laizy — APPROVED** [review]

**Pull request merged**